src/components/views/settings/Notifications.js

```diff
@@ -0,0 +1,1020 @@
+            var rule = this.state.vectorContentRules.rules[i];
+            keywords.push(rule.pattern);
+        }
+        if (keywords.length) {
+            // As keeping the order of per-word push rules hs side is a bit tricky to code,
+            // display the keywords in alphabetical order to the user
+            keywords.sort();
+
+            keywords = keywords.join(", ");
+        }
+        else {
+            keywords = "";
+        }
+
+        var TextInputDialog = sdk.getComponent("dialogs.TextInputDialog");
+        Modal.createDialog(TextInputDialog, {
+            title: "Keywords",
+            description: "Enter keywords separated by a comma:",
+            value: keywords,
+            onFinished: function onFinished(should_leave, newValue) {
+
+                if (should_leave && newValue !== keywords) {
+                    var newKeywords = newValue.split(',');
+                    for (var i in newKeywords) {
+                        newKeywords[i] = newKeywords[i].trim();
+                    }
+
+                    // Remove duplicates and empty
+                    newKeywords = newKeywords.reduce(function(array, keyword){
+                        if (keyword !== "" && array.indexOf(keyword) < 0) {
+                            array.push(keyword);
+                        }
+                        return array;
+                    },[]);
+                    
+                    self._setKeywords(newKeywords);
+                }
+            }
+        });
+    },
+    
+    getRule: function(vectorRuleId) {
+        for (var i in this.state.vectorPushRules) {
+            var rule = this.state.vectorPushRules[i];
+            if (rule.vectorRuleId === vectorRuleId) {
+                return rule;
+            }
+        }
+    },
+    
+    _actionsFor: function(pushRuleVectorState) {
+        if (pushRuleVectorState === PushRuleVectorState.ON) {
+            return ['notify'];
+        }
+        else if (pushRuleVectorState === PushRuleVectorState.LOUD) {
+            return ['notify',
+                {'set_tweak': 'sound', 'value': 'default'},
+                {'set_tweak': 'highlight', 'value': 'true'}
+            ];;
+        }
+    },
+    
+    // Determine whether a rule is in the PushRuleVectorState.ON category or in PushRuleVectorState.LOUD
+    // regardless of its enabled state.
+    _pushRuleVectorStateKind: function(rule) {
+        var stateKind;
+ 
+        // Count tweaks to determine if it is a ON or LOUD rule
+        var tweaks = 0;
+        for (var j in rule.actions) {
+            var action = rule.actions[j];
+            if (action.set_tweak === 'sound' ||
+                (action.set_tweak === 'highlight' &&  action.value)) {
+                tweaks++;
+            }
+        }
+        switch (tweaks) {
+            case 0:
+                stateKind = PushRuleVectorState.ON;
+                break;
+            case 2:
```

Comment:
Well, the method is not so generic as the name suggests.
It is only used for bing words. So, in the specific case of Vector, the code works.
I will rename the method